### PR TITLE
various removal of hardcoded pointers

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

**Comment by BowserSlug** [comment]
⚠️ This PR introduces 37 warnings:
- src/5B320.c:167: warning: initialization makes integer from pointer without a cast
- src/5B320.c:236: warning: initialization makes integer from pointer without a cast
- src/5B320.c:236: warning: initialization makes integer from pointer without a cast
- src/5B320.c:236: warning: initialization makes integer from pointer without a cast
- src/5B320.c:236: warning: initialization makes integer from pointer without a cast
- src/5B320.c:237: warning: initialization makes integer from pointer without a cast
- src/5B320.c:237: warning: initialization makes integer from pointer without a cast
- src/5B320.c:237: warning: initialization makes integer from pointer without a cast
- src/5B320.c:237: warning: initialization makes integer from pointer without a cast
- src/5B320.c:238: warning: initialization makes integer from pointer without a cast
- src/5B320.c:238: warning: initialization makes integer from pointer without a cast
- src/5B320.c:238: warning: initialization makes integer from pointer without a cast
- src/5B320.c:238: warning: initialization makes integer from pointer without a cast
- src/5B320.c:239: warning: initialization makes integer from pointer without a cast
- src/5B320.c:239: warning: initialization makes integer from pointer without a cast
- src/5B320.c:239: warning: initialization makes integer from pointer without a cast
- src/5B320.c:239: warning: initialization makes integer from pointer without a cast
- src/5B320.c:240: warning: initialization makes integer from pointer without a cast
- src/5B320.c:240: warning: initialization makes integer from pointer without a cast
- src/5B320.c:240: warning: initialization makes integer from pointer without a cast
- src/5B320.c:240: warning: initialization makes integer from pointer without a cast
- src/5B320.c:241: warning: initialization makes integer from pointer without a cast
- src/5B320.c:241: warning: initialization makes integer from pointer without a cast
- src/5B320.c:241: warning: initialization makes integer from pointer without a cast
- src/5B320.c:241: warning: initialization makes integer from pointer without a cast
- src/5B320.c:242: warning: initialization makes integer from pointer without a cast
- src/5B320.c:242: warning: initialization makes integer from pointer without a cast
- src/5B320.c:242: warning: initialization makes integer from pointer without a cast
- src/5B320.c:242: warning: initialization makes integer from pointer without a cast
- src/5B320.c:243: warning: initialization makes integer from pointer without a cast
- src/5B320.c:243: warning: initialization makes integer from pointer without a cast
- src/5B320.c:243: warning: initialization makes integer from pointer without a cast
- src/5B320.c:244: warning: initialization makes integer from pointer without a cast
- src/item_hud_script_table.inc.c:1: warning: type defaults to `int' in declaration of `HudScript_MoveDiamond'
- src/item_hud_script_table.inc.c:2: warning: type defaults to `int' in declaration of `HudScript_MoveDiamondDisabled'
- src/item_hud_script_table.inc.c:334: warning: initialization from incompatible pointer type
- src/item_hud_script_table.inc.c:334: warning: initialization from incompatible pointer type

File: src/item_hud_script_table.inc.c [hunk 1]
```diff
@@ -1,3 +1,6 @@
+extern HudScript_MoveDiamond[];
```
Comment (ethteck) [inline src/item_hud_script_table.inc.c:1]: These need types

File: src/world/world.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#ifndef _WORLD_H
+#define _WORLD_H
+
+#include "common.h"
+
+extern const s32 D_800989A0[];
+
+#endif
+
+
```
Comment (ethteck) [inline src/world/world.h:10]: extra newlines

**Review by ethteck — CHANGES_REQUESTED** [review]
nice! also gMessageBoxFrameParts isn't an array of ints, it's at least an array of int pointers. so I'd change its type to s32* [] instead of s32[]

**Review by nanaian — APPROVED** [review]

**Pull request merged**